pkg/aks/create.go

```diff
@@ -59,25 +69,11 @@ func CreateOrUpdateCluster(ctx context.Context, cred *Credentials, clusterClient
 		networkProfile.DNSServiceIP = spec.NetworkDNSServiceIP
 		networkProfile.DockerBridgeCidr = spec.NetworkDockerBridgeCIDR
 		networkProfile.ServiceCidr = spec.NetworkServiceCIDR
-
-		if spec.NetworkPlugin != nil {
-			networkProfile.NetworkPlugin = containerservice.NetworkPlugin(*spec.NetworkPlugin)
-		} else {
-			networkProfile.NetworkPlugin = containerservice.Kubenet
-		}
-
-		// if network plugin is 'Azure', set PodCIDR
-		if networkProfile.NetworkPlugin == containerservice.Azure {
-			networkProfile.PodCidr = spec.NetworkPodCIDR
-		}
-
-		if spec.LoadBalancerSKU != nil {
-			loadBalancerSku := containerservice.LoadBalancerSku(*spec.LoadBalancerSKU)
-			networkProfile.LoadBalancerSku = loadBalancerSku
-		}
-
-		if spec.NetworkPolicy != nil {
-			networkProfile.NetworkPolicy = containerservice.NetworkPolicy(*spec.NetworkPolicy)
+		networkProfile.PodCidr = spec.NetworkPodCIDR
+	} else {
+		// Network Policy Azure can be used only with Network Plugin Azure CNI
+		if containerservice.NetworkPolicy(to.String(spec.NetworkPolicy)) == containerservice.NetworkPolicyAzure {
+			networkProfile.NetworkPolicy = containerservice.NetworkPolicyCalico
```

Comment:
I don't think it's good practice to silently change a configured value behind the user's back like this. It should either be validated and an error returned to let the user correct it, or even just pass the invalid config to AKS and bubble up the error from there.

Reply:
@cmurphy This is fixing an issue with AKSv2 provisioning where the network configuration is not being set properly so the cluster gets deployed with no network policy enabled and kubenet as the default plugin in all cases. As for silently change a configured value, I agree that it should be validated or return an error. The UI has logic to make sure you can't select the Azure network policy without the Azure CNI already.

Reply:
Comments included, thank you